src/SFA.DAS.ApplyService.Web/Infrastructure/Services/ProcessPageFlowService.cs

```diff
@@ -0,0 +1,69 @@
+﻿
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Options;
+using SFA.DAS.ApplyService.Application.Apply.Roatp;
+using SFA.DAS.ApplyService.Domain.Roatp;
+using SFA.DAS.ApplyService.Web.Configuration;
+using SFA.DAS.ApplyService.Web.Infrastructure.Interfaces;
+
+namespace SFA.DAS.ApplyService.Web.Infrastructure.Services
+{
+    public class ProcessPageFlowService : IProcessPageFlowService
+    {
+        private readonly IQnaApiClient _qnaApiClient;
+        private readonly List<TaskListConfiguration> _configuration;
+
+        public ProcessPageFlowService(IQnaApiClient qnaApiClient,
+            IOptions<List<TaskListConfiguration>> configuration)
+        {
+            _qnaApiClient = qnaApiClient;
+            _configuration = _configuration = configuration.Value;
+        }
+
+        public async Task<string> GetIntroductionPageIdForSection(Guid applicationId, int sequenceId,
+            int providerTypeId)
+        {
+            var sequenceDescription = _configuration.FirstOrDefault(x => x.Id == sequenceId);
+
+            if (sequenceDescription == null)
+                return null;
+
+            switch (providerTypeId)
```

Comment:
Rather than having this mapping from id to provider type name, would it be easier to reference the provider type id (1-3) in config?

Reply:
I was avoiding magic numbers by doing this, but I think the code smell cuts both ways.  I think I'll take your suggestion though, I can't think how to make this more elegant